### ISSUE TITLE
Intructions for volumes

### ISSUE THREAD

**Issue by adrianmay** [bkjaya2020/docker-zoneminder-php7.4-mysql8#1]
Please provide instructions for using volumes with this.
It's not much use if it forgets all your footage when the container exits.
I tried 

```
 -v /dockervols/zm/events:/var/cache/zoneminder/events \
 -v /dockervols/zm/mysql:/var/lib/mysql \
 -v /dockervols/zm/log:/var/log/zm \
 -v /dockervols/zm/run:/var/run/zm \
```

and got a sequence of permissions errors which I'm only halfway through resolving.

**Comment by adrianmay** [comment]
That's weird. Closed with no changes to any docs I can see, and no explanation for closing it. Seems like rather odd behaviour. 

If you think it's a stupid issue, at least say why.

**Comment by bkjaya2020** [comment]
Hi brother
Please see the docker file . Opening or comunication with those volumes are
not required for my zoneminder docker repository . Therefore it doesn't
consider as an issue for the repository . Your problem can be solved if you
use sudo chown  www-data:www-data /path to folder

**Comment by bkjaya2020** [comment]
This link may be useful to find a solution for your problem
https://blog.container-solutions.com/understanding-volumes-docker

**Comment by adrianmay** [comment]
I'm looking at the Dockerfile at https://github.com/bkjaya2020/docker-zoneminder-php7.4-mysql8/blob/update1272021/Dockerfile and I still don't see why the data would be persisted. I do see that you run with --privileged which might hold the clue, but it seems like a risk for such a huge software that might have to be exposed to the internet. Other zm images don't seem to need that.

I'm aware of the VOLUME command in Dockerfile which would put the data in some cryptic location, but seeing as CCTV data is huge, you might anticipate people wanting to choose which physical disk to put it on independently of any other docker machines.

I can't see a VOLUME command in the Dockerfile for this repo anyway. I do see some in a related repo called docker-zoneminder-master but I don't see that getting pulled in here. 

I suspect I might have filed this issue under the wrong one of your many zm images, because I must have seen the VOLUME command in order to write the -v parameters I posted in the first place. The article you recommended explains why the permissions weren't set after I applied -v to the volumes, but suggests that everybody could be happy if you used a CMD or ENTRYPOINT instruction to set the permissions.

Besides this, I think you'd agree that none of this is obvious and people might have reasons to override docker's volume storage location. In another application I use lots of volume directories in different states and switch between them. So I still think some changes and doc enhancements would be nice, besides trying to run without --privileged.

Hope that helps,
Adrian.
